model/GameModel.ts

```diff
@@ -1,16 +1,97 @@
+import {resolvePromise} from "./resolvePromise.js";
+
 export class GameModel {
+
+    //celebrity information
     private _name: string;
-    private _imageUrl: string;
-    private _birthYear: number;
-    private _blur: number = 0;
+    private _imageUrl: string = '';
+    private _birth: Hint<Date> = new Hint(1, new Date(0,0,0), true);
+    private _death : Hint<Date> = new Hint(1, new Date(0,0,0));
+    private _occupation : Hint<string>= new Hint(1, '');
+    private _citizenship : Hint<string> = new Hint(1, '');
+    private _initials : Hint<string> = new Hint(2, '');
+    private _paragraph1 : Hint<string> = new Hint (2, '');
+
+    //game information
+    private _blur: number = 4;
+    private _curHintLevel: number = 1;
+    private _hints: Hint<any>[] = [this._death, this._occupation, this._citizenship, this._initials, this._paragraph1];
```

Comment:
Maybe we could only have this list of hints instead of that many attributes. This is not necessary to add them both as attributes and as elements of this list. Maybe we could even create an object which contains that list of hints, i.e. 

```ts
interface HintList {
   birth: Hint<Date>,
   occupation: Hint<string>,
   etc.
}
```

Reply:
I like your idea but we would still need a list or a at least filterable data structure to get random hints, I'm not sure this would work

Reply:
We can easily override the `filter` method of a list to adapt it to our interface, this won't cause any problem.

Reply:
You might want to have a look to my latest commit, I decided to make HintList a class for reasons explained in the commit message, moreover it was way easier to implement a "toList()" function on which we will call filter when needed then write a filter function which would take a predicate as parameter.